mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -485,7 +485,95 @@ LogicalResult tosa::ArgMaxOp::verify() {
   return success();
 }
 
+template <typename T>
```

Comment:
Having some kind of interface here might have helped to avoid templatization.
Nothing to change just an idea.

Reply:
It would likely be good to clean this up in a separate patch since it could also be done for other ops like convolutional ops